hail/python/hail/backend/service_backend.py

```diff
@@ -8,14 +10,43 @@
 
 from hailtop.config import get_deploy_config, get_user_config
 from hailtop.auth import service_auth_headers
-from hailtop.utils import (retry_response_returning_functions,
-                           external_requests_client_session)
+from hailtop.utils import async_to_blocking
 from hail.ir.renderer import CSERenderer
 
 from .backend import Backend
 from ..hail_logging import PythonOnlyLogger
 
 
+class ServiceSocket:
+    def __init__(self, *, deploy_config=None):
+        if not deploy_config:
+            deploy_config = get_deploy_config()
+        self.url = deploy_config.base_url('query')
+        self.session = aiohttp.ClientSession(headers=service_auth_headers(deploy_config, 'query'))
```

Comment:
add a close method to ServiceSocket which closes this session. Call close in backend's stop.